### ISSUE TITLE
[Feature request] Searchable menus

### ISSUE THREAD

**Issue by KasMA1990** [pop-os/cosmic-epoch#757]
Any given application will have some menus at the top of the window. For some, these menus will contain a lot of options, and making them searchable would be really nice. As someone who's recently-ish started using macOS, this is one of those features that are just really nice to have for all applications out of the box, also as something you can find inside the application menus.

The macOS feature is [described here](https://www.howtogeek.com/286168/how-to-search-for-any-menu-option-on-macos/) for example.

**Comment by jacobgkau** [comment]
> Any given application will have some menus at the top of the window.

Many newer apps, especially ones from the GNOME ecosystem, do not-- they may use hamburger menus instead. Some Electron apps also don't have useful ones, since their menus may be implemented within the web engine-driven UI.

This feature request is fine, but it's going to have the same difficulties with implementation that global menus (e.g. always displaying them at the top of the screen like macOS, instead of within the application window) did. From what I recall, when Ubuntu attempted this with Unity, they had to patch some applications to expose the menus to the system. macOS has an easier time since Apple's able to set the standard of how menus are to be implemented within applications.

**Comment by KasMA1990** [comment]
Yeah, and I'm alright with those limitations personally. It does feel best when these things are completely consistent of course, but if the logic is "when top level menu items are available on the widow, then they're searchable" I think that's okay. Even for Electron apps that follow the pattern, but with their own implementation, it'll have to do. Hopefully they look visually distinct enough from the native menus, that it gives the user a hint as to why they don't behave the same.

All in all, I'd rather have this inconsistently than not at all. If COSMIC happens to take over the world (nay, the universe! 😄) then it would suck to sit back and regret not having this very useful UI pattern 😊

**Comment by ikeofkc** [comment]
I think if this is a libcosmic feature and implemented on the right of the menu items as in Mac OS, then it will reduce any maintenance burden for other toolkits as well as make libcosmic applications stand out in quality for professionals looking for a capable application toolkit. 

That being said it would be nice if it support standard d-bus output so that third party solutions would naturally work with it. That way if poeple want to go further to get all their applications searchable with a third party runner type app, then libcosmic apps would not cause issues like GNOME apps do.
